src/checks/exactlyOneOfProperty.ts

```diff
@@ -0,0 +1,50 @@
+import {State, StateMachine, StateMachineError, StateMachineErrorCode} from "../types";
+import {getStates, getStatesContainer, StateEntry} from "./get-states";
+
+export const exactlyOneOfPropertyWorkflow = ({definition, filter, props, errorCode}: {
+    definition: StateMachine,
+    filter: (entry: StateEntry) => boolean
+    props: string[],
+    errorCode: StateMachineErrorCode
+}): StateMachineError[] => {
```

Comment:
refactored the existing terminal check into a common helper function to re-use for other mutual exclusive property checks

using this now with Next/End and WaitDuration